airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/state/GlobalAsyncStateManager.java

```diff
@@ -87,18 +86,19 @@ public class GlobalAsyncStateManager {
    */
   private final ConcurrentMap<Long, AtomicLong> stateIdToCounter = new ConcurrentHashMap<>();
   private final ConcurrentMap<Long, AtomicLong> stateIdToCounterForPopulatingDestinationStats = new ConcurrentHashMap<>();
-  private final ConcurrentMap<Long, ImmutablePair<PartialAirbyteMessage, Long>> stateIdToState = new ConcurrentHashMap<>();
+  private final ConcurrentMap<Long, ImmutablePair<StateMessageWithArrivalNumber, Long>> stateIdToState = new ConcurrentHashMap<>();
```

Comment:
how is the arrival number different from the state id?

Reply:
Arrival number specifically denotes the order in which the state arrived. Its main purpose is just for logging so that its easier to debug if the destination is emitting the states in the same order as it received it.